hudi-common/src/main/java/org/apache/hudi/metadata/SecondaryIndexRecordGenerationUtils.java

```diff
@@ -275,25 +198,39 @@ public static HoodieData<HoodieRecord> readSecondaryKeysFromFileSlices(HoodieEng
       } else {
         readerSchema = tableSchema;
       }
-      return createSecondaryIndexGenerator(metaClient, engineType, logFilePaths, readerSchema, partition, dataFilePath, indexDefinition,
-          metaClient.getActiveTimeline().filterCompletedInstants().lastInstant().map(HoodieInstant::requestedTime).orElse(""));
+      ClosableIterator<Pair<String, String>> secondaryIndexGenerator = createSecondaryIndexGenerator(readerContextFactory.getContext(), metaClient, fileSlice, readerSchema, indexDefinition,
+          metaClient.getActiveTimeline().filterCompletedInstants().lastInstant().map(HoodieInstant::requestedTime).orElse(""), props, false);
+      return new CloseableMappingIterator<>(secondaryIndexGenerator, pair -> createSecondaryIndexRecord(pair.getKey(), pair.getValue(), indexDefinition.getIndexName(), false));
     });
   }
 
-  private static ClosableIterator<HoodieRecord> createSecondaryIndexGenerator(HoodieTableMetaClient metaClient,
-                                                                              EngineType engineType, List<String> logFilePaths,
-                                                                              Schema tableSchema, String partition,
-                                                                              Option<StoragePath> dataFilePath,
-                                                                              HoodieIndexDefinition indexDefinition,
-                                                                              String instantTime) throws Exception {
-    return new ClosableIterator<HoodieRecord>() {
-      private final HoodieFileSliceReader<HoodieRecord> fileSliceReader = getFileSliceReader(
-          metaClient, engineType, logFilePaths, tableSchema, partition, dataFilePath, instantTime);
-      private HoodieRecord nextValidRecord;
+  private static <T> ClosableIterator<Pair<String, String>> createSecondaryIndexGenerator(HoodieReaderContext<T> readerContext,
```

Comment:
I know this method was not added in this patch. 
but can we rename this to `createSecondaryIndexRecordGenerator`

Reply:
Yes, updated